src/compiler/proposals-and-stabilization/custom-test-jobs.md

```diff
@@ -0,0 +1,96 @@
+
+- may spuriously fail; or
+- may genuinely fail but it's not clear who should be consulted about the failure or who is responsible for fixing the failure; or
+- otherwise do not have well documented test job maintainers and failure protocol;
+
+can introduce a lot of friction and frustration for other contributors who now will have to deal with the failure or understand why the test job failed in the first place, which may now be blocking their PR. Everyone utilizing the [rust-lang/rust] CI is expected to use it responsibly.
+
+To help with this, please follow the process described below if you'd like to propose adding a custom test job/component to the [rust-lang/rust] CI, and copy and fill out the Custom Test Job/Component Policy.
+
+## Process for adding a custom test job to rust-lang/rust CI
+
+![Flowchart of the process for adding a custom test job/component to rust-lang/rust CI](./custom-test-jobs/test-job-flow.svg)
+
+## Custom Test Job/Component Template
+
+*Please copy and fill out this template as part of the MCP.*
+
+The proposed custom test job or custom test component for the [rust-lang/rust] CI must:
+
+- Be approved by the compiler team through a proposed Major Change Proposal (MCP), where the MCP is seconded by a compiler team member, and the MCP being accepted with no blocking concerns.
+- Have no blocking concerns from the library team by creating an issue on [rust-lang/rust] proposing to add the custom test job/component to [rust-lang/rust] CI linking to the MCP, and nominating the issue for review by the library team via `@rustbot label +I-libs-nominated`.
+- Have the implementation PR be reviewed and approved by the infrastructure team, linking to the MCP and issue.
+- Be properly documented on [rustc-dev-guide] (preferably as part of the implementation PR), linking to the MCP and issue.
+
+### Test job/component rationale
+
+- What does this custom test job/component do?
+- What [rust-lang/rust] changes can potentially break the custom test job/component? E.g. changes to rustc, standard library, bootstrap or tools (like clippy/rustfmt/cargo).
+- Why does this custom test job/component need to be part of the [rust-lang/rust] PR and/or Full Merge CI?
+- If the custom test job/component will block on failure, why does it need to block?
+- If the custom test job/component will not block on failure initially but is intended to eventually become blocking, why so? And when will it become blocking?
+
+### Test job/component maintainers
+
+The proposed custom test job/component for [rust-lang/rust] CI must have at least one dedicated maintainer. The test job/component maintainers understand that they will be pinged or otherwise contacted about the custom test job/component, particularly for (but not limited to) its failures. Please list who will be maintaining this test job/component here.
```

Comment:
Discussion: we could also be very upfront about what happens if a custom test job/component no longer having a dedicated maintainer, that:

- We may ask around if anyone wants to sign up as a dedicated maintainer for this specific test job/component
- If we can't find a dedicated maintainer, and that e.g. compiler/libs/infra (or if the test job can break tools in r-l/r, e.g. cargo/rustdoc/...) find the test job/component to not pull its weight / cause significant burden / is no longer relevant, then the test job/component may be removed.

Reply:
I think that would be a reasonable addition.

Reply:
Added this remark about unmaintained job/components may be removed in (round 2).